evals/elsuite/audio/eval.py

```diff
@@ -47,82 +51,64 @@ def __init__(
         self.text_only = text_only
         self.temperature = temperature
         self.max_tokens = max_tokens
+        self._recorder = None
+
+    @property
+    def recorder(self):
+        return self._recorder
 
     def eval_sample(self, sample: Sample, rng):
         assert isinstance(sample, Sample)
-        task_prompt = self.build_prompt(sample)
-        if self.text_only:
-            content = task_prompt.replace(AUDIO_PLACEHOLDER, sample.transcript)
-        else:
-            content = []
-            with BytesIO() as buffer:
-                sf.write(buffer, sample.audio, DEFAULT_SAMPLE_RATE, format="WAV", subtype="PCM_16")
-                base_64_wav = base64.b64encode(buffer.getvalue()).decode()
-            parts = task_prompt.split(AUDIO_PLACEHOLDER)
-            assert len(parts) > 1
-            if parts[0]:
-                content.append({"type": "text", "text": parts[0]})
-            content.append(
-                {
-                    "type": "image_url",
-                    "image_url": {
-                        "url": f"data:audio/x-wav;base64,{base_64_wav}",
-                    },
-                }
-            )
-            if parts[1]:
-                content.append({"type": "text", "text": parts[1]})
+        prompt = self.build_prompt(sample, self.text_only)
+        kwargs = self.get_completion_kwargs(sample)
+        sampled = self.do_completion(prompt, **kwargs)
+        return self.compute_metrics(sample, sampled)
+
+    def run(self, recorder: RecorderBase):
+        samples = self.load_dataset()
+        self._recorder = recorder
+        self.eval_all_samples(recorder, samples)
+        return self.compute_corpus_metrics()
+
+    def load_dataset(self):
+        ds = load_hf_dataset(self.dataset, evals.eval._MAX_SAMPLES).cast_column(
+            "audio", Audio(sampling_rate=DEFAULT_SAMPLE_RATE)
+        )
```

Comment:
yeah we can do this already, in the tools eval I override this because the `cast_column` ends up being different. LMK if you had something else in mind.